### PR TITLE
Test an update to MPICH 4.0.3

### LINKED ISSUE #22993: Update conda package MPICH to 4.0.3
## Reason
[MPICH 4.0.3 was recently released](https://www.mpich.org/2022/11/08/mpich-4-0-3-released/) with several improvements and various bugfixes. We should make this available to our users.

## Design
Create a `moose-mpich` package using version 4.0.3 and propagate it up the `moose-*` conda packages that use it.

## Impact
Add new bugfixes and features to `moose-mpich`

### PR DESCRIPTION
**Pull request by cticenhour**
Bumping to 4.0.3 means bumping clang to 14.0.6 with GCC remaining at 10.4.0.

Closes #22993

### CODE REVIEW

**Review by milljm — DISMISSED** [review]
I'd call this a win, if you wanted to go ahead and merge...

**Comment by cticenhour** [comment]
I definitely tried to update as few builds/versions as was required, so I might have missed something. Might need to resort to a `mamba create -n test <install_all_the_packages_ever needed>` mega command and double-check through things. The last issues I had locally were with `tools` / `test-tools` dependencies, so perhaps start there.

**Comment by loganharbour** [comment]
Can we use this as an opportunity to fix the stuff?

**Comment by milljm** [comment]
@loganharbour I don't see why not!

@cticenhour Obviously locally... everything installs as it should. So I am a little confused why Civet is the only thing failing. For exactness... Here is the install order that Civet is executing for the following targets (since we know package placement matters):

```pre
❯ grep "^CIVET_CONDA_PACKAGES" *
Conda.cfg:CIVET_CONDA_PACKAGES = moose-petsc moose-libmesh-vtk python=3.7 moose-tools
Create_Arm_Packages.cfg:CIVET_CONDA_PACKAGES = pyyaml jinja2
Create_Linux_Packages.cfg:CIVET_CONDA_PACKAGES = pyyaml jinja2
Create_Mac_Packages.cfg:CIVET_CONDA_PACKAGES = pyyaml jinja2
Mac_LibTorch.cfg:CIVET_CONDA_PACKAGES = moose-tools moose-libmesh
Mac_arm_LibTorch.cfg:CIVET_CONDA_PACKAGES = moose-tools moose-libmesh
Python310_Tests.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-libmesh-vtk moose-tools python=3.10
Python37_Tests.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-libmesh-vtk moose-tools python=3.7
Python38_Tests.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-libmesh-vtk moose-tools python=3.8
Python39_Tests.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-libmesh-vtk moose-tools python=3.9
Use_Mac_Packages.cfg:CIVET_CONDA_PACKAGES = moose-petsc moose-libmesh-vtk moose-tools
end_user-arm64.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-tools python=3.10
end_user-arm64_extended.cfg:CIVET_CONDA_PACKAGES = moose-mpich moose-tools python=3.10
end_user-centos.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-tools python=3.10
end_user-mac.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-tools python=3.10
end_user-mac_extended.cfg:CIVET_CONDA_PACKAGES = moose-mpich moose-tools python=3.10
end_user-rocky_extended.cfg:CIVET_CONDA_PACKAGES = moose-petsc moose-tools
end_user-ubuntu.cfg:CIVET_CONDA_PACKAGES = moose-libmesh moose-tools python=3.10
end_user-ubuntu_extended.cfg:CIVET_CONDA_PACKAGES = moose-petsc moose-tools
```

**Comment by milljm** [comment]
🤦 yep
```bash
mamba create -n testing moose-libmesh moose-libmesh-vtk moose-tools python=3.9 openssl=1 
<trimmed>
  + moose-libmesh                2022.11.07  build_2               conda.software.inl.gov/public/osx-64               34MB
  + moose-libmesh-vtk                 9.1.0  hd77c2c2_16           conda.software.inl.gov/public/osx-64               30MB
  + moose-mpich                       4.0.2  build_6               conda.software.inl.gov/public/osx-64                7kB
  + moose-petsc                      3.16.6  build_5               conda.software.inl.gov/public/osx-64               28MB
  + moose-test-tools             2022.12.15  py39hea4d9ce_0        conda.software.inl.gov/moose/pr-22977/osx-64        4kB
  + moose-tools                  2022.12.05  py39hea4d9ce_0        conda.software.inl.gov/public/osx-64                4kB
```

**Comment by cticenhour** [comment]
@loganharbour go nuts if you need to use this branch for something else. Started out as just a testing branch, but if we're OK with bumping MPICH to 4.0.3 and clang to 14, then go for it.

@milljm Oh yeah, that would do it. We don't need that hack any longer right?

File: conda/mpich/conda_build_config.yaml [hunk 1]
```diff
@@ -1,42 +1,34 @@
...
-  - mpich-mpicxx 4.0.2                                     # [linux]
-  - mpich-mpicxx 4.0.2                                     # [not arm64 and osx]
-  - mpich-mpicxx 4.0.2                                     # [arm64]
+  - mpich-mpicxx 4.0.3
 
 base_mpifort:
-  - mpich-mpifort 4.0.2                                     # [linux]
-  - mpich-mpifort 4.0.2                                     # [not arm64 and osx]
-  - mpich-mpifort 4.0.2                                     # [arm64]
+  - mpich-mpifort 4.0.3
```
Comment (milljm) [inline conda/mpich/conda_build_config.yaml:12]: 👍

**Comment by cticenhour** [comment]
Closing this for a revisit later on. This updates clang to a version we don't necessarily need to go to yet (which obviously introduces some test failures), for very little gain on the side of MPICH.

**Pull request closed without merging**